### PR TITLE
Update mutation v6

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

Adds translation for update mutations, filters and projection

### CODE REVIEW

File: packages/graphql/src/api-v6/queryIR/UpdateOperation.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            throw new Error("No parent node found!");
+        }
+
+        const { selection: selectionClause, nestedContext } = this.selection.apply(context);
+
+        const setParams = this.inputFields.flatMap((p) => p.getSetParams(nestedContext));
+
+        selectionClause.set(...setParams);
+
+        // TODO: filter subqueries
```
Comment (MacondoExpress) [inline packages/graphql/src/api-v6/queryIR/UpdateOperation.ts:76]: any reason why the getSubqueries is not called?

File: packages/graphql/src/api-v6/queryIRFactory/resolve-tree-parser/argument-parser/parse-update-args.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export function parseUpdateOperationArgsTopLevel(resolveTreeArgs: Record<string, any>): GraphQLTreeUpdate["args"] {
+    return {
+        input: parseUpdateOperationInput(resolveTreeArgs.input),
+        where: resolveTreeArgs.where,
+    };
+}
+
+function parseUpdateOperationInput(resolveTreeCreateInput: any): GraphQLTreeUpdateInput[] {
+    if (!resolveTreeCreateInput || !Array.isArray(resolveTreeCreateInput)) {
```
Comment (MacondoExpress) [inline packages/graphql/src/api-v6/queryIRFactory/resolve-tree-parser/argument-parser/parse-update-args.ts]: this should be changed, contains `create` references, the input should not be an array

File: packages/graphql/src/api-v6/queryIRFactory/resolve-tree-parser/graphql-tree/graphql-tree.ts [hunk 3]
```diff
@@ -33,6 +37,14 @@ export interface GraphQLTreeCreate extends GraphQLTreeNode {
     };
 }
 
+export interface GraphQLTreeUpdate extends GraphQLTreeNode {
+    name: string;
+    args: {
+        where: GraphQLWhereTopLevel;
+        input: GraphQLTreeUpdateInput[];
```
Comment (MacondoExpress) [inline packages/graphql/src/api-v6/queryIRFactory/resolve-tree-parser/graphql-tree/graphql-tree.ts]:
```suggestion
        input: GraphQLTreeUpdateInput;
```

File: packages/graphql/src/api-v6/translators/translate-update-operation.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+export function translateUpdateOperation({
+    context,
+    entity,
+    graphQLTreeUpdate,
+}: {
+    context: Neo4jGraphQLTranslationContext;
+    graphQLTreeUpdate: GraphQLTreeUpdate;
+    entity: ConcreteEntity;
+}): Cypher.CypherResult {
+    const createFactory = new UpdateOperationFactory(context.schemaModel);
```
Comment (MacondoExpress) [inline packages/graphql/src/api-v6/translators/translate-update-operation.ts]: wrong variable name

File: packages/graphql/tests/api-v6/integration/mutations/update/update.int.test.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            mutation {
+                ${Movie.operations.update}(
+                    where: {
+                        node: {
+                            title: {
+                                equals: "The Matrix"
+                            }
+                        }
+                    }
+                    input: [
```
Comment (MacondoExpress) [inline packages/graphql/tests/api-v6/integration/mutations/update/update.int.test.ts]: Input should be an Object

**Review by MacondoExpress — COMMENTED** [review]
Left some comments! Do you want to open a follow-up PR for the tests, like Spatial Types / Temporals / `@alias`?

File: packages/graphql/src/api-v6/queryIR/UpdateOperation.ts (hunk 1 above)
Comment (angrykoala) [reply packages/graphql/src/api-v6/queryIR/UpdateOperation.ts:76]: At the moment I'm not sure of an example using them (as we don't support all features yet) and I'm not too happy adding this code until we can test it

File: packages/graphql/src/api-v6/queryIRFactory/resolve-tree-parser/argument-parser/parse-update-args.ts (hunk 2 above)
Comment (angrykoala) [reply packages/graphql/src/api-v6/queryIRFactory/resolve-tree-parser/argument-parser/parse-update-args.ts:31]: Why not? update input supports an array

File: packages/graphql/src/api-v6/queryIR/UpdateOperation.ts (hunk 1 above)
Comment (angrykoala) [reply packages/graphql/src/api-v6/queryIR/UpdateOperation.ts:76]: I'm not sure of the example in which this is used, as we don't have all features yet, and I'm not comfortable adding untested code

**Pull request merged**